spec/support/content_data_api.rb

```diff
@@ -10,6 +10,12 @@ def content_data_api_has_metric(base_path:, from:, to:, metrics:, payload:)
         stub_request(:get, url).to_return(status: 200, body: body)
       end
 
+      def content_data_api_does_not_have_metric(base_path:, from:, to:, metrics:)
+        query = GdsApi::ContentDataApi.new.query(from: from, to: to, metrics: metrics)
```

Comment:
Not sure if the parameters: `from`, `to` and `metrics` should be in the signature of this method, because not having metrics between those dates would not imply having the error. The `404` should only happen if the `base_path` is not present. 
I would set up the fixture with:

```ruby
content_data_api_does_not_have_metric(base_path: 'base/path')
```

Reply:
You need them in there to construct the correct URL to stub, I'm not sure of another way to stub this - any ideas gratefully received.

Reply:
Renamed the method to `content_data_api_does_not_have_base_path`